std/typecons.d

```diff
@@ -4628,6 +4640,65 @@ unittest
     //func(n);
 }
 
+///The proxied value must be an $(B lvalue).
+unittest
+{
+    struct NewIntType
+    {
+        //Won't work; the literal '1' is
+        //is an rvalue, not an lvalue
+        //mixin Proxy!1; 
+        
+        //Okay, n is an lvalue
+        int n;
+        mixin Proxy!n;
+        
+        this(int n) { this.n = n; }
+    }
+    
+    NewIntType nit = 0;
+    nit++;
+    assert(nit == 1);
+    
```

Comment:
What about showing that the proxied type is not implicitly convertible to/from the base value, as well as constitute a distinct type?